app/models/offer.rb

```diff
@@ -119,6 +119,15 @@ def scheduled?
     end
   end
 
+  def self.find_by_ggv_uuid(ggv_uuid)
+    begin
+      offer = GogovanOrder.find_by(ggv_uuid: ggv_uuid).delivery.offer
+    rescue Exception => e
+      raise ActiveRecord::RecordNotFound
+    end
+    with_eager_load.find(offer.id)
+  end
```

Comment:
Good point re 404. Also, we only want to show offers when the GGV order is in a valid collection state, either pending or active. Rather than catch the exception I think it's better to use ```try``` here as it is more specific. 

    offer = GogovanOrder.find_by_ggv_uuid(params[:ggv_uuid]).try(:delivery).try(:offer)
    raise ActiveRecord::RecordNotFound unless offer